pydra/utils/hash.py

```diff
@@ -224,10 +224,27 @@ def bytes_repr_dict(obj: dict, cache: Cache) -> Iterator[bytes]:
     yield b"}"
 
 
-@register_serializer(_SpecialForm)
+@register_serializer(ty._GenericAlias)
+@register_serializer(ty._SpecialForm)
 @register_serializer(type)
 def bytes_repr_type(klass: type, cache: Cache) -> Iterator[bytes]:
-    yield f"type:({klass.__module__}.{klass.__name__})".encode()
+    try:
+        yield f"type:({klass.__module__}.{klass.__name__}".encode()
+    except AttributeError:
+        yield f"type:(typing.{klass._name}:(".encode()  # type: ignore
+    args = ty.get_args(klass)
+    if args:
```

Comment:
Why not create a new dispatch method for the Generic/SpecialForm cases, rather than all this branching logic?

Reply:
I tried a few different things but it is a bit tricky because typing special forms/generics aren't stable across Python 3.8-3.11. In later versions they appear like a proper class but in earlier versions they are more like an object. L234 doesn't even work for Python 3.8 & 3.9, as `klass._name` is None for Union.

In the typing module I ended up having to pull in the obscure `typing_utils` module to get typing to work, so I might need to do something similar here